app/models/game.rb

```diff
@@ -7,5 +7,19 @@ class Game < ActiveRecord::Base
   
   has_many :pieces
   has_many :players
+
+  def piece_present?(x, y)
+    if self.pieces.where(x_position: x, y_position: y).any?
+      return true
+    else
+      return false
+    end
+  end
+
+  def get_piece(x, y)
+    if self.piece_present?(x, y)
+      return self.pieces.where(x_position: x, y_position: y).take
```

Comment:
Why double up the query by checking if the piece is present first? `.take` will return empty recordset if no results, which IIRC evaluates to a false in non-type strict expressions